rdmo/projects/progress.py

```diff
@@ -8,42 +8,49 @@
 from rdmo.questions.models import Page, Question, QuestionSet
 
 
-def resolve_conditions(project, values):
-    # get all conditions for this catalog
-    pages_conditions_subquery = Page.objects.filter_by_catalog(project.catalog) \
+def get_catalog_conditions(catalog):
+    pages_conditions_subquery = Page.objects.filter_by_catalog(catalog) \
                                             .filter(conditions=OuterRef('pk'))
-    questionsets_conditions_subquery = QuestionSet.objects.filter_by_catalog(project.catalog) \
+    questionsets_conditions_subquery = QuestionSet.objects.filter_by_catalog(catalog) \
                                                           .filter(conditions=OuterRef('pk'))
-    questions_conditions_subquery = Question.objects.filter_by_catalog(project.catalog) \
+    questions_conditions_subquery = Question.objects.filter_by_catalog(catalog) \
                                                     .filter(conditions=OuterRef('pk'))
 
-    catalog_conditions = Condition.objects.annotate(has_page=Exists(pages_conditions_subquery)) \
-                                          .annotate(has_questionset=Exists(questionsets_conditions_subquery)) \
-                                          .annotate(has_question=Exists(questions_conditions_subquery)) \
-                                          .filter(Q(has_page=True) | Q(has_questionset=True) | Q(has_question=True)) \
-                                          .distinct().select_related('source', 'target_option')
+    return Condition.objects.annotate(
+        has_page=Exists(pages_conditions_subquery),
+        has_questionset=Exists(questionsets_conditions_subquery),
+        has_question=Exists(questions_conditions_subquery)
+    ).filter(
+        Q(has_page=True) | Q(has_questionset=True) | Q(has_question=True)
+    ).distinct().select_related('source', 'target_option')
+
+
+def resolve_conditions(catalog, values):
+    # resolve conditions and return for each condition the set_indexes which resolved true
+    conditions = defaultdict(set)
+    for condition in get_catalog_conditions(catalog):
+        resolved_set_indexes = {value.set_index for value in values if condition.resolve([value])}
```

Comment:
no, this fails with
```py
>                   assert page['show'] == show, page['uri']
E                   AssertionError: http://example.com/terms/questions/catalog/conditions/text_empty
E                   assert True == False
```
The previous ways works

```py
    conditions = defaultdict(set)
    for condition in get_catalog_conditions(catalog):
        if condition.resolve(values):
            resolved_value_set_indexes = {value.set_index for value in values if condition.resolve([value])}
            conditions[condition.id].update(resolved_value_set_indexes)
```

Reply:
Yes, I see, but it does not really makes sense to first check with all values and then check each value individually.